drivers/volume/kdmp/kdmp.go

```diff
@@ -231,26 +239,174 @@ func (k *kdmp) UpdateMigratedPersistentVolumeSpec(
+	// TODO: need to make sure pv name remains updated
+	for _, object := range objects {
+		var pvc v1.PersistentVolumeClaim
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(object.UnstructuredContent(), &pvc); err != nil {
+			continue
+		}
+		sc := k8shelper.GetPersistentVolumeClaimClass(&pvc)
+		if val, ok := restore.Spec.StorageClassMapping[sc]; ok {
+			pvc.Spec.StorageClassName = &val
+			pvc.Spec.VolumeName = ""
+		}
+		if pvc.Annotations != nil {
+			delete(pvc.Annotations, "pv.kubernetes.io/bind-completed")
+			delete(pvc.Annotations, "pv.kubernetes.io/bound-by-controller")
+		}
+		o, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&pvc)
+		if err != nil {
+			logrus.Errorf("unable to pvc to unstruct objects, err: %v", err)
+			return nil, err
+		}
+		object.SetUnstructuredContent(o)
+		pvcs = append(pvcs, object)
+	}
+	return pvcs, nil
 }
 
 func (k *kdmp) StartRestore(
 	restore *storkapi.ApplicationRestore,
 	volumeBackupInfos []*storkapi.ApplicationBackupVolumeInfo,
 ) ([]*storkapi.ApplicationRestoreVolumeInfo, error) {
+	log.ApplicationRestoreLog(restore).Debugf("started generic restore: %v", restore.Name)
 	volumeInfos := make([]*storkapi.ApplicationRestoreVolumeInfo, 0)
+	for _, bkpvInfo := range volumeBackupInfos {
+		volumeInfo := &storkapi.ApplicationRestoreVolumeInfo{}
+		// wait for pvc to get bound
+		err := wait.ExponentialBackoff(volumeAPICallBackoff, func() (bool, error) {
+			pvc, err := core.Instance().GetPersistentVolumeClaim(bkpvInfo.PersistentVolumeClaim, bkpvInfo.Namespace)
+			if err != nil {
+				return false, nil
+			}
+			if pvc.Spec.VolumeName == "" {
+				return false, nil
+			}
+			volumeInfo.RestoreVolume = pvc.Spec.VolumeName
+			return true, nil
+		})
+		if err != nil {
+			logrus.Errorf("Failed to get volume restore volume: %s, err: %v", bkpvInfo.PersistentVolumeClaim, err)
+			continue
+		}
+		if err != nil {
+			return nil, err
+		}
+		volumeInfo.PersistentVolumeClaim = bkpvInfo.PersistentVolumeClaim
+		volumeInfo.SourceNamespace = bkpvInfo.Namespace
+		volumeInfo.SourceVolume = bkpvInfo.Volume
+		volumeInfo.DriverName = driverName
+
+		// create VolumeBackup CR
+		volBackup := &kdmpapi.VolumeBackup{}
+		volBackup.Annotations = make(map[string]string)
+		volBackup.Annotations[skipResourceAnnotation] = "true"
+		volBackup.Name = getGenericCRName(prefixRestore+"-"+restore.Name, bkpvInfo.Namespace, bkpvInfo.PersistentVolumeClaim)
+		volBackup.Namespace = bkpvInfo.Namespace
+		volBackup.Spec.BackupLocation = kdmpapi.DataExportObjectReference{
+			Kind:       reflect.TypeOf(storkapi.BackupLocation{}).Name(),
+			Name:       restore.Spec.BackupLocation,
+			Namespace:  restore.Namespace,
+			APIVersion: StorkAPIVersion,
+		}
+		volBackup.Spec.Repository = fmt.Sprintf("%s/%s-%s/", prefixRepo, bkpvInfo.Namespace, bkpvInfo.PersistentVolumeClaim)
+		volBackup.Status.SnapshotID = bkpvInfo.BackupID
+		if _, err := kdmpShedOps.Instance().CreateVolumeBackup(volBackup); err != nil {
```

Comment:
Should we check for already exist error and ignore the error, when any retry happens for a volume.